js/demo_policy/main.ts

```diff
@@ -0,0 +1,202 @@
+  );
+}
+
+/**
+ * Handles OCI manifest digest lookup failure scenario
+ */
+function handleOciManifestDigestFailure(): Validation.ValidationResponse {
+  const image = 'registry.testing.lan/nonexistent-image:1.0.0';
+  let digest: string | null = null;
+  try {
+    digest = ManifestDigest.getOCIManifestDigest(image);
+  } catch (e) {
+    console.error('OCI manifest digest lookup failed:', e);
+    return new Validation.ValidationResponse(
+      false,
+      `OCI manifest digest lookup failed: ${e}`,
+      undefined,
+      undefined,
+      { digest: '' }
+    );
+  }
+  // If digest is empty, treat as failure
+  return new Validation.ValidationResponse(
+    false,
+    'Unexpectedly retrieved OCI manifest digest',
+    undefined,
+    undefined,
+    { digest: digest }
+  );
+}
+
+/**
+ * Handles DNS lookup success scenario
+ */
+function handleDnsLookupSuccess(): Validation.ValidationResponse {
+  let ips: string[] = [];
+  try {
+    ips = Network.dnsLookup('google.com').ips || [];
+  } catch (e) {
+    console.error('DNS lookup failed:', e);
+  }
+  return new Validation.ValidationResponse(
+    !!ips && ips.length > 0,
+    ips && ips.length > 0 ? undefined : 'Failed to retrieve DNS lookup IPs',
+    undefined,
+    undefined,
+    { ips: ips.join(', ') || '' }
+  );
+}
+
+/**
+ * Handles DNS lookup failure scenario
+ */
+function handleDnsLookupFailure(): Validation.ValidationResponse {
+  let ips: string[] = [];
+  try {
+    ips = Network.dnsLookup('invalid.nonexistent.tld').ips || [];
+    if (ips.length > 0) {
+      return new Validation.ValidationResponse(
+        false,
+        'Unexpectedly retrieved DNS lookup IPs',
+        undefined,
+        undefined,
+        { ips: ips.join(', ') }
+      );
+    } else {
+      return new Validation.ValidationResponse(
+        false,
+        'DNS lookup returned no results',
+        undefined,
+        undefined,
+        { ips: '' }
+      );
+    }
+  } catch (e) {
+    console.error('DNS lookup failed:', e);
+    return new Validation.ValidationResponse(
+      false,
+      `DNS lookup failed: ${e}`,
+      undefined,
+      undefined,
+      { ips: '' }
+    );
+  }
+}
+
+/**
+ * Handles the default privileged container validation
+ */
+function handlePrivilegedContainerValidation(validationRequest: any, settings: PolicySettings): Validation.ValidationResponse {
+  const pod = JSON.parse(JSON.stringify(validationRequest.request.object)) as Pod;
+  const privileged =
+    pod.spec?.containers?.some(container => container.securityContext?.privileged) || false;
+
+  if (privileged) {
+    if (settings.ignoredNamespaces?.includes(validationRequest.request.namespace || '')) {
+      console.error('privileged container are allowed inside of ignored namespace');
+      return Validation.acceptRequest();
+    } else {
+      return Validation.rejectRequest('privileged containers are not allowed');
+    }
+  } else {
+    return Validation.acceptRequest();
+  }
```

Comment:
Maybe I was the one who wrote this code, but I don't like it now :smile:  I don't like all these `if/else/else`.

It could be rewritten as:


```ts
if (privileged) {
  if (settings.ignoredNamespaces?.includes(validationRequest.request.namespace || '')) {
    console.error('privileged container are allowed inside of ignored namespace');
    return Validation.acceptRequest();
  }
    
  return Validation.rejectRequest('privileged containers are not allowed');
}

return Validation.acceptRequest();
```

Also, we could even check immediately whether the request is coming from an ignored namespace and return immediately, without looking for the privileged container